test/Data/PreciseDateTime.purs

```diff
@@ -47,40 +47,41 @@ spec =
         `shouldEqual` (Just $ preciseDateTimeFixture 0 0)
 
       fromRFC3339String (RFC3339String $ dateStringFixture <> ".1Z")
-        `shouldEqual` (Just $ preciseDateTimeFixture 100 100000000)
+        `shouldEqual` (Just $ preciseDateTimeFixture 100 0)
 
       fromRFC3339String (RFC3339String $ dateStringFixture <> ".01Z")
-        `shouldEqual` (Just $ preciseDateTimeFixture 10 10000000)
+        `shouldEqual` (Just $ preciseDateTimeFixture 10 0)
 
       fromRFC3339String (RFC3339String $ dateStringFixture <> ".001Z")
-        `shouldEqual` (Just $ preciseDateTimeFixture 1 1000000)
+        `shouldEqual` (Just $ preciseDateTimeFixture 1 0)
 
       fromRFC3339String (RFC3339String $ dateStringFixture <> ".10Z")
-        `shouldEqual` (Just $ preciseDateTimeFixture 100 100000000)
+        `shouldEqual` (Just $ preciseDateTimeFixture 100 0)
 
       fromRFC3339String (RFC3339String $ dateStringFixture <> ".100Z")
-        `shouldEqual` (Just $ preciseDateTimeFixture 100 100000000)
+        `shouldEqual` (Just $ preciseDateTimeFixture 100 0)
 
       fromRFC3339String (RFC3339String $ dateStringFixture <> ".123Z")
-        `shouldEqual` (Just $ preciseDateTimeFixture 123 123000000)
+        `shouldEqual` (Just $ preciseDateTimeFixture 123 0)
 
       fromRFC3339String (RFC3339String $ dateStringFixture <> ".999999999Z")
-        `shouldEqual` (Just $ preciseDateTimeFixture 999 999999999)
+        `shouldEqual` (Just $ preciseDateTimeFixture 999 999999)
 
       fromRFC3339String (RFC3339String $ dateStringFixture <> ".000000001Z")
         `shouldEqual` (Just $ preciseDateTimeFixture 0 1)
 
+      -- Take the first 9 digits if more than 9 are provided
       fromRFC3339String (RFC3339String $ dateStringFixture <> ".1000000000Z")
-        `shouldEqual` Nothing
+        `shouldEqual` (Just $ preciseDateTimeFixture 100 0)
```

Comment:
@natefaubion this is the adjusted test case that I think may address https://github.com/awakesecurity/purescript-precise-datetime/issues/8.